### PR TITLE
CLDC-2220 Update buyer pluralisation

### PR DESCRIPTION
**Pull request by kosiakkatrina**

### CODE REVIEW

File: app/models/form/sales/pages/housing_benefits.rb [hunk 1]
```diff
@@ -14,7 +14,7 @@ def depends_on
     if @joint_purchase
       [{ "joint_purchase?" => true }]
     else
-      [{ "not_joint_purchase?" => true }]
+      [{ "not_joint_purchase?" => true }, { "jointpur" => nil }]
```
Comment (kosiakkatrina) [inline app/models/form/sales/pages/housing_benefits.rb:17]: Joint purchase is routed to conditionally, so might not always be answered, added this here just to be sure we have default question version. Any reason we shouldn't do that?

Comment (natdeanlewissoftwire) [reply app/models/form/sales/pages/housing_benefits.rb:17]: Hm, good point - still displaying this question for nil jointpurs makes sense to me - will this make any previously complete logs become in progress if they hadn't satisfied the condition for jointpur to be routed to?

File: app/models/form/sales/questions/privacy_notice.rb [hunk 2]
```diff
@@ -1,19 +1,20 @@
 class Form::Sales::Questions::PrivacyNotice < ::Form::Question
-  def initialize(id, hsh, page)
-    super
+  def initialize(id, hsh, page, joint_purchase:)
+    super(id, hsh, page)
     @id = "privacynotice"
-    @check_answer_label = "Buyer has seen the privacy notice?"
+    @check_answer_label = "#{joint_purchase ? 'Buyers have' : 'Buyer has'} seen the privacy notice?"
```
Comment (natdeanlewissoftwire) [inline app/models/form/sales/questions/privacy_notice.rb:5]: non-technical question - how are we expecting users to answer if one but not all buyers (in a joint purchase) have seen the privacy notice?

File: app/models/form/sales/questions/savings.rb [hunk 3]
```diff
@@ -1,8 +1,8 @@
 class Form::Sales::Questions::Savings < ::Form::Question
-  def initialize(id, hsh, page)
-    super
+  def initialize(id, hsh, page, joint_purchase:)
+    super(id, hsh, page)
     @id = "savings"
-    @check_answer_label = "Buyer’s total savings before any deposit paid"
+    @check_answer_label = "#{joint_purchase ? 'Buyers’' : 'Buyer’s'} total savings before any deposit paid"
```
Comment (natdeanlewissoftwire) [inline app/models/form/sales/questions/savings.rb:5]: there's some inconsistency of `’` vs `'` in this PR (though I accept there is throughout the codebase too) - maybe a tech debt ticket to sort this out later as it's such a non-issue?

Comment (natdeanlewissoftwire) [reply app/models/form/sales/questions/savings.rb:5]: (looks like we're using the first one in "buyer's" and the second one in "weren't" for example)

File: app/models/validations/sales/financial_validations.rb [hunk 4]
```diff
@@ -48,7 +48,7 @@ def validate_percentage_bought_not_greater_than_percentage_owned(record)
     return unless record.stairbought && record.stairowned
 
     if record.stairbought > record.stairowned
-      record.errors.add :stairowned, I18n.t("validations.financial.staircasing.percentage_bought_must_be_greater_than_percentage_owned")
+      record.errors.add :stairowned, I18n.t("validations.financial.staircasing.percentage_bought_must_be_greater_than_percentage_owned", persons: record.joint_purchase? ? "buyers" : "buyer")
```
Comment (natdeanlewissoftwire) [inline app/models/validations/sales/financial_validations.rb]: can we call `persons` something like `buyers_with_apostrophe`? I think `persons` doesn't clarify whether it's about "more than one person" or "person's"

File: app/models/validations/sales/financial_validations.rb [hunk 5]
```diff
@@ -125,9 +125,9 @@ def validate_equity_less_than_staircase_difference(record)
 
     if record.equity > record.stairowned - record.stairbought
       formatted_equity = sprintf("%g", record.equity)
-      record.errors.add :equity, I18n.t("validations.financial.equity.over_stairowned_minus_stairbought", equity: formatted_equity, staircase_difference: record.stairowned - record.stairbought)
-      record.errors.add :stairowned, I18n.t("validations.financial.equity.over_stairowned_minus_stairbought", equity: formatted_equity, staircase_difference: record.stairowned - record.stairbought)
-      record.errors.add :stairbought, I18n.t("validations.financial.equity.over_stairowned_minus_stairbought", equity: formatted_equity, staircase_difference: record.stairowned - record.stairbought)
+      record.errors.add :equity, I18n.t("validations.financial.equity.over_stairowned_minus_stairbought", equity: formatted_equity, staircase_difference: record.stairowned - record.stairbought, person_owns: record.joint_purchase? ? "buyers own" : "buyer owns")
```
Comment (natdeanlewissoftwire) [inline app/models/validations/sales/financial_validations.rb]: similary can we call this buyers_own_with_apostrophe (or just buyers_own would be good)?

File: app/models/validations/sales/property_validations.rb [hunk 6]
```diff
@@ -4,10 +4,10 @@ def validate_postcodes_match_if_discounted_ownership(record)
     return unless record.ppostcode_full.present? && record.postcode_full.present?
 
     if record.discounted_ownership_sale? && record.ppostcode_full != record.postcode_full
-      record.errors.add :postcode_full, I18n.t("validations.property.postcode.must_match_previous")
-      record.errors.add :ppostcode_full, I18n.t("validations.property.postcode.must_match_previous")
-      record.errors.add :ownershipsch, I18n.t("validations.property.postcode.must_match_previous")
-      record.errors.add :uprn, I18n.t("validations.property.postcode.must_match_previous")
+      record.errors.add :postcode_full, I18n.t("validations.property.postcode.must_match_previous", persons: record.joint_purchase? ? "Buyers‘" : "Buyer‘s")
```
Comment (natdeanlewissoftwire) [inline app/models/validations/sales/property_validations.rb]: same as above

**Review by natdeanlewissoftwire — APPROVED** [review]
nice!

File: app/models/form/sales/questions/privacy_notice.rb (hunk 2 above)
Comment (kosiakkatrina) [reply app/models/form/sales/questions/privacy_notice.rb:5]: Good point, will double check

File: app/models/form/sales/questions/savings.rb (hunk 3 above)
Comment (kosiakkatrina) [reply app/models/form/sales/questions/savings.rb:5]: Fixed it for weren't/wasn't, good spot! Don't think this PR updated buyers ones!

**Pull request merged**